code/game/objects/structures/watercloset.dm

```diff
@@ -480,6 +459,175 @@
+			flick("baton_active", src)
+			user.Paralyze(B.stun_time)
+			user.stuttering = B.stun_time/20
+			B.deductcharge(B.cell_hit_cost)
+			user.visible_message("<span class='warning'>[user] shocks [user.p_them()]self while attempting to wash the active [B.name]!</span>", \
+								"<span class='userdanger'>You unwisely attempt to wash [B] while it's still on.</span>")
+			playsound(src, B.stun_sound, 50, TRUE)
+			return
+
+	if(istype(O, /obj/item/mop))
+		O.reagents.add_reagent(dispensedreagent, 5)
+		to_chat(user, "<span class='notice'>You wet [O] in [src].</span>")
+		playsound(loc, 'sound/effects/slosh.ogg', 25, TRUE)
+		return
+
+	if(O.tool_behaviour == TOOL_WRENCH && !(flags_1&NODECONSTRUCT_1))
+		O.play_tool_sound(src)
+		deconstruct()
+		return
+
+	if(istype(O, /obj/item/stack/medical/gauze))
+		var/obj/item/stack/medical/gauze/G = O
+		new /obj/item/reagent_containers/glass/rag(src.loc)
+		to_chat(user, "<span class='notice'>You tear off a strip of gauze and make a rag.</span>")
+		G.use(1)
+		return
+
+	if(istype(O, /obj/item/stack/ore/glass))
+		new /obj/item/stack/sheet/sandblock(loc)
+		to_chat(user, "<span class='notice'>You wet the sand in the sink and form it into a block.</span>")
+		O.use(1)
+		return
+
+	if(!istype(O))
+		return
+	if(O.item_flags & ABSTRACT) //Abstract items like grabs won't wash. No-drop items will though because it's still technically an item in your hand.
+		return
+
+	if(user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>You start washing [O]...</span>")
+		busy = TRUE
+		if(!do_after(user, 40, target = src))
+			busy = FALSE
+			return 1
+		busy = FALSE
+		O.wash(CLEAN_WASH)
+		O.acid_level = 0
+		create_reagents(5)
+		reagents.add_reagent(dispensedreagent, 5)
+		reagents.expose(O, TOUCH)
+		user.visible_message("<span class='notice'>[user] washes [O] using [src].</span>", \
+							"<span class='notice'>You wash [O] using [src].</span>")
+		return 1
+	else
+		return ..()
+
+/obj/structure/legacysink/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		drop_materials()
+	..()
+
+/obj/structure/legacysink/proc/drop_materials()
+	if(buildstacktype)
+		new buildstacktype(loc,buildstackamount)
+	else
+		for(var/i in custom_materials)
+			var/datum/material/M = i
+			new M.sheet_type(loc, FLOOR(custom_materials[M] / MINERAL_MATERIAL_AMOUNT, 1))
+
+/obj/structure/legacysink/puddle	//splishy splashy ^_^
+	name = "puddle"
+	desc = "A puddle used for washing one's hands and face."
+	icon_state = "puddle"
+	resistance_flags = UNACIDABLE
+
+//ATTACK HAND IGNORING PARENT RETURN VALUE
+/obj/structure/legacysink/puddle/attack_hand(mob/M)
+	icon_state = "puddle-splash"
+	. = ..()
+	icon_state = "puddle"
+
+/obj/structure/legacysink/puddle/attackby(obj/item/O, mob/user, params)
+	icon_state = "puddle-splash"
+	. = ..()
+	icon_state = "puddle"
+
+/obj/structure/legacysink/puddle/deconstruct(disassembled = TRUE)
+	qdel(src)
+
+/obj/structure/legacysink/greyscale
```

Comment:
this is only useful for crafting/creating material datum based sinks, seeing as the use-case for these isn't crafted sinks you dont need a greyscale type